tests/unit/development/metadata_test.py

```diff
@@ -91,7 +92,31 @@ class Table(Component):
     _namespace = 'TableComponents'
     _type = 'Table'
     @_explicitize_args
-    def __init__(self, children=None, optionalArray=Component.UNDEFINED, optionalBool=Component.UNDEFINED, optionalFunc=Component.UNDEFINED, optionalNumber=Component.UNDEFINED, optionalObject=Component.UNDEFINED, optionalString=Component.UNDEFINED, optionalSymbol=Component.UNDEFINED, optionalNode=Component.UNDEFINED, optionalElement=Component.UNDEFINED, optionalMessage=Component.UNDEFINED, optionalEnum=Component.UNDEFINED, optionalUnion=Component.UNDEFINED, optionalArrayOf=Component.UNDEFINED, optionalObjectOf=Component.UNDEFINED, optionalObjectWithExactAndNestedDescription=Component.UNDEFINED, optionalObjectWithShapeAndNestedDescription=Component.UNDEFINED, optionalAny=Component.UNDEFINED, customProp=Component.UNDEFINED, customArrayProp=Component.UNDEFINED, id=Component.UNDEFINED, **kwargs):
+    def __init__(
+        self,
+        children: typing.Union[str, int, float, Component, typing.List[typing.Union[str, int, float, Component]]] = None,
+        optionalArray: typing.List = Component.UNDEFINED,
```

Comment:
this should be a `List[Any]`. mypy will throw an error on this. 

also, wondering if we should use the parent type `Iterable`.

Reply:
`typing.List` is same as `typing.List[typing.Any]`, iterable might not json compatible, I think the List is the most safe for this typing.

Reply:
There are certainly a number of other iterables people frequently use though - `tuple`, `range`, `dict_keys`, `dict_values`, etc etc...  I wouldn't want to push people into casting everything to a list, that would be unnecessary overhead.

Reply:
Still open 😎

Reply:
We can add Tuple as union for untyped arrays, but `arrayOf` is really a different typing on both side.

Reply:
It's better to go without defining type aliases that would require to be imported and would break backward compatibility for component libraries that compiled with the new typing but the user has dash locked to a previous version.